bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/logic/Event.java

```diff
@@ -66,12 +69,58 @@ public boolean equals(@Nullable Object other) {
             return false;
         }
         final Event otherEvent = (Event) other;
-        return (this.title.equals(otherEvent.title) && this.start.equals(otherEvent.start)
-                && this.end.equals(otherEvent.end));
+        final String title = this.title;
+        final String comment = this.comment;
+        final String contact = this.contact;
+        final String description = this.description;
+        final String location = this.location;
+        return (this.start.equals(otherEvent.start) && this.end.equals(otherEvent.end)
+                && ((title == null && otherEvent.title == null) || (title != null && title.equals(otherEvent.title)))
+                && ((comment == null && otherEvent.comment == null)
+                        || (comment != null && comment.equals(otherEvent.comment)))
+                && ((contact == null && otherEvent.contact == null)
+                        || (contact != null && contact.equals(otherEvent.contact)))
+                && ((description == null && otherEvent.description == null)
+                        || (description != null && description.equals(otherEvent.description)))
+                && ((location == null && otherEvent.location == null)
+                        || (location != null && location.equals(otherEvent.location))));
+    }
+
+    @Override
+    public int hashCode() {
+        final String title = this.title;
+        final String comment = this.comment;
+        final String contact = this.contact;
+        final String description = this.description;
+        final String location = this.location;
+        return this.start.hashCode() + this.end.hashCode() + (title != null ? title.hashCode() : 0)
```

Comment:
Can be simplified just as `return Objects.hash(this.start, this.end, this.title, this.description, this.location, this.comment, this.contact)`. `java.util.Objects#hash` already does the null check for us.

Reply:
Nice hint.